File: FoodOrderApis/OrderService/Data/Models/Order.cs
using OrderService.Enums;

namespace OrderService.Data.Models;

public class Order
{
    public int Id { get; set; }
    public DateTime OrderedDate { get; set; }
    public int OrderStatus { get; set; }
    public string EaterId { get; set; }
    public User Eater { get; set; }
    public string MerchantId { get; set; }
    public User Merchant { get; set; }
    public List<OrderDetail> OrderDetails { get; set; }
}

File: food_order_app/lib/data/models/order.dart
import 'package:food_order_app/data/models/brief_user.dart';

class Order {
  int id;
  DateTime? orderedDate;
  int orderStatus;
  String? eaterId;
  String? merchantId;
  BriefUser? eater;
  BriefUser? merchant;
  Order(
      {required this.id,
      this.orderedDate,
      required this.orderStatus,
      this.eaterId,
      this.merchantId,
      this.merchant,
      this.eater});

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
        id: json['id'],
        orderStatus: json['orderStatus'],
        eaterId: json['eaterId'],
        merchantId: json['merchantId'],
        orderedDate: json['orderedDate'] != null
            ? DateTime.parse(json['orderedDate'])
            : null,
        eater: json['eater'] != null ? BriefUser.fromJson(json['eater']) : null,
        merchant: json['merchant'] != null
            ? BriefUser.fromJson(json['merchant'])
            : null);
  }
}
